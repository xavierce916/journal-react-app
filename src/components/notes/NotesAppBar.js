import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {
    
    const dispatch = useDispatch();
    const { active } = useSelector( ({ notes }) => notes );
    
    const handleSave = () => {
        dispatch( startSaveNote( active ) );
    }

    const handlePictureCLick = () => {
        document.querySelector('#fileSelector').click()    
    }

    const handleFileChange = (e) => {
        const file = e.target.files[0];
        console.log(file);
        if ( file ) {
            dispatch( startUploading( file ) );
            document.querySelector('#fileSelector').value = '';
        }
    }

    return (
        <div className="notes__appbar">
            <span>28 de agosto 2022</span>

            <input
                id="fileSelector"
                type="file"
                name="file"
                style={{ display: 'none' }}
                onChange={ handleFileChange }
            />

            <div>
                <button 
                    className="btn"
                    onClick={ handlePictureCLick }
                >
                    Picture
                </button>

                <button 
                    className="btn"
                    onClick={ handleSave }
                >
                    Save
                </button>
            </div>
        </div>
    );
};
