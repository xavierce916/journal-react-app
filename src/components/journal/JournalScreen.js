import { NothingSelected } from "./NothingSelected";
import { NoteScreen } from "../notes/NoteScreen";
import { Sidebar } from "./Sidebar";
import { useSelector } from "react-redux";

export const JournalScreen = () => {

    const { active } = useSelector( ({ notes }) => notes );
    

    return (
        <div className="journal__main-content animate__animated animate__fadeIn animate__faster">

            <Sidebar />

            <main>
                {
                    ( !!active )
                       ? ( <NoteScreen /> )
                       : ( <NothingSelected /> )
                }
                

            </main>
        </div>    
    )
  };