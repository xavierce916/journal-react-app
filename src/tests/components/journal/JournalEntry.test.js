import { mount } from 'enzyme';
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
    id: '',
    url: 'https://foto.png'
}

const wrapper = mount( 
    <Provider store={ store }>
        <JournalEntry { ...note }  /> 
    </Provider>

);

describe('Pruebas en <JournalEntry />', () => {

    test('debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe activar la nota', () => {

        wrapper.find( '.journal__entry' ).prop('onClick')();

        expect( store.dispatch ).toHaveBeenCalledWith(
            activeNote( note.id, { ...note } )
        )
    });
});