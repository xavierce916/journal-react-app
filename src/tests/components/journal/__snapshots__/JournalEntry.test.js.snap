// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry /> debe mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url( https://foto.png )",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    />
    <p
      className="jornal__entry-content"
    />
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Tuesday
    </span>
    <h4>
      1st
    </h4>
  </div>
</div>
`;
